matching-notification-api/src/main/java/org/phenotips/matchingnotification/storage/internal/DefaultMatchStorageManager.java

```diff
@@ -563,21 +592,41 @@ public boolean setUserContacted(PatientMatch match, boolean isUserContacted)
     }
 
     @Override
-    public boolean updateNotificationHistory(PatientMatch match, JSONObject notificationRecord)
+    public Collection<PatientMatch> updateNotificationHistory(PatientMatch match, JSONObject notificationRecord)
     {
         Session session = this.beginTransaction();
         boolean transactionCompleted = false;
 
+        Collection<PatientMatch> matchesToUpdate = new ArrayList<>();
+        matchesToUpdate.add(match);
+
+        if (!match.isLocal()) {
+            String referenceServerId = match.isIncoming() ? match.getReferenceServerId() : "";
+            String matchedServerId = match.isIncoming() ? "" : match.getMatchedServerId();
+
+            // Load equivalent remote match
+            PatientMatch equivalentMatch = loadMatchBetweenPatients(match.getMatchedPatientId(),
+                matchedServerId, match.getReferencePatientId(), referenceServerId);
+            if (equivalentMatch != null) {
+                matchesToUpdate.add(equivalentMatch);
+            }
+        }
+
+        Collection<PatientMatch> updatedMatches = new ArrayList<>();
         try {
-            match.updateNotificationHistory(notificationRecord);
-            session.update(match);
-            transactionCompleted = true;
+            for (PatientMatch matchToUpdate : matchesToUpdate) {
+                matchToUpdate.updateNotificationHistory(notificationRecord);
+                session.update(matchToUpdate);
+                transactionCompleted = true;
```

Comment:
`transactionCompleted = true;` should only be used if/when all matches were updated, not just one of many in the loop. If any one failed, we should roll back the entire transaction